xbmc/filesystem/CurlFile.cpp

```diff
@@ -601,9 +601,15 @@ void CCurlFile::SetCommonOptions(CReadState* state, bool failOnError /* = true *
       StringUtils::Format(":%d", m_proxyport);
     g_curlInterface.easy_setopt(h, CURLOPT_PROXY, hostport.c_str());
 
-    const std::string userpass =
-      m_proxyuser + std::string(":") + m_proxypassword;
-    if (userpass != ":")
+    std::string userpass = "";
```

Comment:
No need to add `= ""`. It will initialise to its default constructor as it's an object.